### PR TITLE
Add cupy.average

### PR DESCRIPTION
**Pull request by rezoo**
This PR implements `cupy.average`.

### CODE REVIEW

File: cupy/statistics/meanvar.py [hunk 1]
```diff
@@ -1,7 +1,73 @@
...
+        returned (bool): If ``True``, a tuple of the average and the sum
+            of weights is returned, otherwise only the average is returned.
+
+    Returns:
+        cupy.ndarray or tuple of cupy.ndarray: The average of the input array
+            along the axis and the sum of weights.
+
+    .. seealso:: :func:`numpy.average`
+    """
+    a = cupy.asarray(a)
```
Comment (okuta) [inline cupy/statistics/meanvar.py]: How about using `isinstance` ?

File: cupy/statistics/meanvar.py [hunk 2]
```diff
@@ -1,7 +1,73 @@
...
+
+    if weights is None:
+        avg = a.mean(axis)
+        scl = avg.dtype.type(a.size / avg.size)
+    else:
+        wgt = cupy.asarray(weights)
```
Comment (okuta) [inline cupy/statistics/meanvar.py]: ditto.

**Comment by rezoo** [comment]
I fixed.

**Comment by rezoo** [comment]
After the discussion we decided not to use `isinstance()`.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**